x/emissions/keeper/actor_penalties.go

```diff
@@ -0,0 +1,126 @@
+package keeper
+
+import (
+	"errors"
+
+	"cosmossdk.io/collections"
+	errorsmod "cosmossdk.io/errors"
+	alloraMath "github.com/allora-network/allora-chain/math"
+	"github.com/allora-network/allora-chain/x/emissions/types"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+)
+
+// UpdateTopicInfererPenalties updates the penalty for the given topic based on the current EMA scores of the current active set.
+func (k *Keeper) UpdateTopicInfererPenalties(ctx sdk.Context, topicId TopicId) error {
+	lowestEMAScore, exists, err := k.GetLowestInfererScoreEma(ctx, topicId)
+	if err != nil {
+		return errorsmod.Wrap(err, "error getting lowest inferer score EMA")
+	}
+
+	// Empty active set so no penalty
+	if !exists {
+		err = k.penaltyInTopic.Remove(ctx, topicId)
+		if err != nil {
+			return errorsmod.Wrap(err, "error removing penalty for topic")
+		}
+	}
+
+	// Get the EMA score from the active set
+	var scores []alloraMath.Dec
+	if err := k.activeInferers.Walk(
+		ctx,
+		collections.NewPrefixedPairRange[TopicId, ActorId](topicId),
+		func(key collections.Pair[TopicId, ActorId]) (stop bool, err error) {
+			score, err := k.GetInfererScoreEma(ctx, topicId, key.K2())
+			if err != nil {
+				return false, errorsmod.Wrap(err, "error getting inferer score EMA")
+			}
+			scores = append(scores, score.Score)
+			return false, nil
+		},
+	); err != nil {
+		return errorsmod.Wrap(err, "error walking active inferers")
+	}
+
+	params, err := k.GetParams(ctx)
+	if err != nil {
+		return errorsmod.Wrap(err, "error getting params")
+	}
+	standardDeviation, _, err := alloraMath.MedianAbsoluteDeviation(scores)
+	if err != nil {
+		return errorsmod.Wrap(err, "error calculating median absolute deviation")
+	}
+	x, err := standardDeviation.Mul(params.SortitionLambdaPenalty)
```

Comment:
change x for a more meaningful name (unless there is a justification on keeping name from paper equations?)